apps/meteor/tests/end-to-end/api/livechat/contacts.ts

```diff
@@ -0,0 +1,277 @@
+import { createUser } from '../../../data/users.helper';
+
+describe('LIVECHAT - contacts', () => {
+	before((done) => getCredentials(done));
+
+	before(async () => {
+		await updateSetting('Livechat_enabled', true);
+		await updatePermission('create-livechat-contact', ['admin']);
+	});
+
+	after(async () => {
+		await restorePermissionToRoles('create-livechat-contact');
+		await updateSetting('Livechat_enabled', false);
+	});
+
+	describe('[POST] omnichannel/contacts', () => {
+		it('should be able to create a new contact', async () => {
+			const res = await request
+				.post(api('omnichannel/contacts'))
+				.set(credentials)
+				.send({
+					name: faker.person.fullName(),
+					emails: [faker.internet.email().toLowerCase()],
+					phones: [faker.phone.number()],
+				});
+
+			expect(res.body).to.have.property('success', true);
+			expect(res.body).to.have.property('contactId');
+			expect(res.body.contactId).to.be.an('string');
+		});
+
+		it("should return an error if user doesn't have 'create-livechat-contact' permission", async () => {
+			await removePermissionFromAllRoles('create-livechat-contact');
+
+			const res = await request
+				.post(api('omnichannel/contacts'))
+				.set(credentials)
+				.send({
+					name: faker.person.fullName(),
+					emails: [faker.internet.email().toLowerCase()],
+					phones: [faker.phone.number()],
+				});
+
+			expect(res.body).to.have.property('success', false);
+			expect(res.body.error).to.be.equal('User does not have the permissions required for this action [error-unauthorized]');
+
+			await restorePermissionToRoles('create-livechat-contact');
+		});
+
+		it('should return an error if contact manager not exists', async () => {
+			const res = await request
+				.post(api('omnichannel/contacts'))
+				.set(credentials)
+				.send({
+					name: faker.person.fullName(),
+					emails: [faker.internet.email().toLowerCase()],
+					phones: [faker.phone.number()],
+					contactManager: 'invalid',
+				});
+
+			expect(res.body).to.have.property('success', false);
+			expect(res.body).to.have.property('error');
+			expect(res.body.error).to.be.equal('error-contact-manager-not-found');
+		});
+
+		it('should return an error if contact manager is not a livechat-agent', async () => {
+			const normalUser = await createUser();
+
+			const res = await request
+				.post(api('omnichannel/contacts'))
+				.set(credentials)
+				.send({
+					name: faker.person.fullName(),
+					emails: [faker.internet.email().toLowerCase()],
+					phones: [faker.phone.number()],
+					contactManager: normalUser._id,
+				});
+
+			expect(res.body).to.have.property('success', false);
+			expect(res.body).to.have.property('error');
+			expect(res.body.error).to.be.equal('error-invalid-contact-manager');
+		});
+
+		it('should be able to create a new contact with a contact manager', async () => {
+			const livechatAgent = await createAgent();
```

Comment:
Can we create a new user for this instead of using the admin one? And after all delete that new user?